File: mobile-client/android/Contract/System/Entitlement/EntitlementSnapshot.kt
package app.mobiling.client.contract.system.entitlement

/**
 * Canonical contract snapshot for client entitlement state.
 */
data class EntitlementSnapshot(
    val subjectId: String,
    val grantedCodes: List<String> = emptyList(),
    val refreshedAtIso8601: String? = null,
)

File: mobile-client/ios/Contract/System/Entitlement/EntitlementSnapshot.swift
import Foundation

public struct EntitlementSnapshot: Sendable, Equatable {
    public let subjectId: String
    public let grantedCodes: [String]
    public let refreshedAtIso8601: String?

    public init(subjectId: String, grantedCodes: [String] = [], refreshedAtIso8601: String? = nil) {
        self.subjectId = subjectId
        self.grantedCodes = grantedCodes
        self.refreshedAtIso8601 = refreshedAtIso8601
    }
}
